cmd/app/add.go

```diff
@@ -85,38 +87,50 @@ func preRunAddCommand(ctx context.Context, clients *shared.ClientFactory) error
 	if !clients.Config.WithExperimentOn(experiment.BoltFrameworks) {
 		return nil
 	}
-	manifestSource, err := clients.Config.ProjectConfig.GetManifestSource(ctx)
-	if err != nil {
-		return err
-	}
-	if manifestSource.Equals(config.ManifestSourceRemote) {
-		return slackerror.New(slackerror.ErrAppInstall).
-			WithMessage("Apps cannot be installed due to project configurations").
-			WithRemediation(
-				"Install an app on app settings: %s\nLink an app to this project with %s\nList apps saved with this project using %s",
-				style.LinkText("https://api.slack.com/apps"),
-				style.Commandf("app link", false),
-				style.Commandf("app list", false),
-			).
-			WithDetails(slackerror.ErrorDetails{
-				slackerror.ErrorDetail{
-					Code:    slackerror.ErrProjectConfigManifestSource,
-					Message: "Cannot install apps with manifests sourced from app settings",
-				},
-			})
-	}
 	return nil
 }
 
 // RunAddCommand executes the workspace install command, prints output, and returns any errors.
 func RunAddCommand(ctx context.Context, clients *shared.ClientFactory, selection *prompts.SelectedApp, orgGrantWorkspaceID string) (context.Context, types.InstallState, types.App, error) {
 	if selection == nil {
-		selected, err := teamAppSelectPromptFunc(ctx, clients, prompts.ShowHostedOnly, prompts.ShowAllApps)
+		var isProductionApp bool
+		var err error
+
+		// When team flag is provided, default app environment to deployed if not specified.
+		// TODO(semver:major): Remove defaulting to deployed and require the environment flag to be set.
+		if clients.Config.TeamFlag != "" && addFlags.environmentFlag == "" {
+			addFlags.environmentFlag = "deployed"
+			isProductionApp = true
+			clients.IO.PrintDebug(ctx,
+				"Defaulting app environment to deployed because team flag is provided. "+
+					"Please use '--environment deployed' to avoid breaking changes in the next major version.",
+			)
+		} else {
+			// Prompt for deployed or local app environment.
+			isProductionApp, err = promptIsProduction(ctx, clients)
+			if err != nil {
+				return ctx, "", types.App{}, err
+			}
+		}
```

Comment:
question: I really don't like this implementation and would appreciate suggestions.

Today, `slack install` always uses a deployed app. So scripts using `slack install --team T0123` always create and install a deployed app .

This PR introduces an app environment prompt, which will block the scripts. To keep backwards compatibility, we default to deployed when no `--environment` is specified.

The `promptIsProduction` supports `clients.Config.Flags.Lookup("environment")`. However, I'm having trouble having this flag listed and right now it always return `nil`. If we can set the flag to deployed, this code can become less nested and more linear to read.